### ISSUE TITLE
Running isml lint on new project throws error.

### ISSUE THREAD

**Issue by bogdan-pomirleanu** [FabiowQuixada/isml-linter#4]
The "eslint-to-isscript" rule is enabled, but an ESLint configuration ".eslintrc.json" file could not be found in the project root directory.
✖ Failed to run ismllint!
{ message:
   'No eslint configuration found. Please add an ESLint configuration file and try again.',
  isCustom: true }

Seems is looking for **.eslintrc.json** but it should also accept **.eslintrc** file. For projects which are not using the .json extension.

**Comment by FabiowQuixada** [comment]
Hello @bogdan-pomirleanu , thanks for you suggestion.

While this isn't available, you can use the "eslintConfig" configuration to point to your .eslintrc file.

Regards,

**Comment by rcambrj** [comment]
It looks like `ConfigUtils.loadEslintConfig` is expecting `eslintConfigParam` but never receives it. It also looks like this is the `--eslintConfig` parameter. So the only way to use this linter is with `.eslintrc.json`.

https://github.com/FabiowQuixada/isml-linter/blob/a255935902f4a41cee1f826a606b167650581bfc/src/util/ConfigUtils.js#L50
https://github.com/FabiowQuixada/isml-linter/blob/a255935902f4a41cee1f826a606b167650581bfc/src/rules/tree/eslint-to-isscript.js#L34

**Comment by rcambrj** [comment]
My bad, I now see that the `eslintConfig` option should be defined in `ismllinter.config.js`, but that also appears to be ignored.

This configuration doesn't throw an error.
```
const config = {
  eslingConfig: 'this_file_doesnt_exist.json',
  rules: {
    ...
  }
};

module.exports = config;
```

But if I remove the configuration file, the linter complains about its absence.

**Comment by fabioquixadaosf** [comment]
Hello @rcambrj , thanks for your feedback and sorry for not been clear on my first comment about where the configuration should be set.

I assume you tried:

`eslingConfig: '.eslintrc',`

and you have that file under your root directory, correct? If so, I will check what might be happening.

Meanwhile, please take a careful look at the rule's documentation,to make sure to side effect is introduced: https://github.com/FabiowQuixada/isml-linter/blob/HEAD/docs/rules/eslint-to-isscript.md

I'll also take a look on the non-thrown error you mentioned.

Thanks!

**Comment by rcambrj** [comment]
@fabioquixadaosf the test case is:
1. point the config at a non-existent `eslintrc` (doesn't complain that the `eslintrc` doesn't exist
2. remove the isml-linter config (complains that the isml-linter config doesn't exist)

So, we know that it's reading this config file, but for some reason ignoring the `eslintConfig` option.

Furthermore, I discovered just now that the `rootDir` option is also ignored, and here's why:

https://github.com/FabiowQuixada/isml-linter/blob/master/bin/isml-linter.js#L16
https://github.com/FabiowQuixada/isml-linter/blob/a255935902f4a41cee1f826a606b167650581bfc/src/publicApi.js#L15
https://github.com/FabiowQuixada/isml-linter/blob/a255935902f4a41cee1f826a606b167650581bfc/src/Builder.js#L6
https://github.com/FabiowQuixada/isml-linter/blob/a255935902f4a41cee1f826a606b167650581bfc/src/IsmlLinter.js#L149

In the first file, the parameter `path` is not specified for `IsmlLinter.build()` and that defaults, in the last file, to `appRoot.toString()`, so running as `npx isml-linter --build` will always ignore the config option `rootDir`.

**Comment by FabiowQuixada** [comment]
@rcambrj ,
I did a quick test on my side, and it seems there is a misspelling on the configuration you used:

`eslingConfig: 'this_file_doesnt_exist.json',`

Please use "eslintConfig" instead and check if it works as expected. I'll soon add a validation on configuration attributes to avoid such typo issues.

Thanks!

**Comment by rcambrj** [comment]
OMG that's embarassing. Can confirm, the `eslintConfig` option is respected. Sorry!

Can't say the same thing for `rootDir` though :(

**Comment by FabiowQuixada** [comment]
No worries :)

If it happened to you, it could happen to anyone else, so I'll add the config attributes validation soon, as I mentioned.

I will check the `rootDir` configuration :)

Once again, thanks!

**Comment by FabiowQuixada** [comment]
Hello @rcambrj ,

the "rootDir" config issue should be fixed by now, please check if it works as expected. Furthermore, if you have anything else related to it, please report it under `[this issue](https://github.com/FabiowQuixada/isml-linter/issues/5 )`. 

Thank you!

**Comment by FabiowQuixada** [comment]
Hello @bogdan-pomirleanu ,

the requested feature has been implemented and is now available, starting from version 5.22.0.

Please update Isml Linter and provide a feedback if it works as expected.

Thank you,

**Comment by FabiowQuixada** [comment]
Will close this issue due to inactivity.
